File: mobile/src/main/java/treehou/se/habit/main/MainActivityComponent.java
package treehou.se.habit.main;


import dagger.Subcomponent;
import treehou.se.habit.module.ActivityComponent;
import treehou.se.habit.module.ActivityComponentBuilder;
import treehou.se.habit.module.ActivityScope;

@ActivityScope
@Subcomponent(
        modules = MainActivityModule.class
)
public interface MainActivityComponent extends ActivityComponent<MainActivity> {

    @Subcomponent.Builder
    interface Builder extends ActivityComponentBuilder<MainActivityModule, MainActivityComponent> {
    }
}

File: mobile/src/main/java/treehou/se/habit/ui/main/MainActivityComponent.kt
package treehou.se.habit.ui.main


import dagger.Subcomponent
import treehou.se.habit.module.ActivityComponent
import treehou.se.habit.module.ActivityComponentBuilder
import treehou.se.habit.module.ActivityScope

@ActivityScope
@Subcomponent(modules = arrayOf(MainActivityModule::class))
interface MainActivityComponent : ActivityComponent<MainActivity> {

    @Subcomponent.Builder
    interface Builder : ActivityComponentBuilder<MainActivityModule, MainActivityComponent>
}